packages/ember-glimmer/lib/component.js

```diff
@@ -117,6 +117,30 @@ const Component = CoreView.extend(
       }
     },
 
+    /**
+      You must call `destroy` on a view to destroy the view (and all of its
+      child views). This will remove the view from any parent node, then make
+      sure that the DOM element managed by the view can be released by the
+      memory manager.
+
+      @method destroy
+      @private
+    */
+    destroy() {
+      // get parentView before calling super because it'll be destroyed
+      let parentView = this.parentView;
+      let viewName = this.viewName;
+
+      if (!this._super(...arguments)) { return; }
+
+      // remove from non-virtual parent view if viewName was specified
+      if (viewName && parentView) {
+        parentView.set(viewName, null);
```

Comment:
I am not sure we actually support this anymore.

@krisselden - What do you think?